File: app/src/main/java/com/example/recyclerviewapp/CustomViewModel.java
package com.example.recyclerviewapp;

public class CustomViewModel {

    final int icon;
    final String name;
    SwipeState state;

    public CustomViewModel(int icon, String name) {
        this.icon = icon;
        this.name = name;
        this.state = SwipeState.NONE;
    }

    public CustomViewModel(int icon, String name, SwipeState state) {
        this.icon = icon;
        this.name = name;
        this.state = state;
    }
}

File: app/src/main/java/com/example/recyclerviewapp/CustomViewModel.kt
package com.example.recyclerviewapp

class CustomViewModel {

    var icon: Int? = null
    var name: String? = null

    constructor(icon: Int, name: String) {
        this.icon = icon
        this.name = name
    }

    companion object {
        const val SWIPE_NONE = 0
        const val SWIPE_LEFT = 1
        const val SWIPE_RIGHT = 2
        const val SWIPE_LEFT_RIGHT = 3
    }
}
